From a1fd2056b0e3820920778fd9252cec645cbb6389 Mon Sep 17 00:00:00 2001
From: Michael Lee <michael-cy.lee@mediatek.com>
Date: Fri, 7 Jul 2023 17:14:40 +0800
Subject: [PATCH 37/40] hostapd: mtk: Fix wpa_supplicant configuration parsing
 error

In the original flow, after hostapd_config_tx_queue successfully
parses a tx_queue variable, it would not return immediately. Then it
would print out "unknow global field" later and set return val to -1.

This patch returns immediately after hostapd_config_tx_queue
successfully parses a tx_queue variable.

Signed-off-by: Michael Lee <michael-cy.lee@mediatek.com>
---
 wpa_supplicant/config.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/wpa_supplicant/config.c b/wpa_supplicant/config.c
index 325ab8d..2652c8a 100644
--- a/wpa_supplicant/config.c
+++ b/wpa_supplicant/config.c
@@ -5715,6 +5715,7 @@ int wpa_config_process_global(struct wpa_config *config, char *pos, int line)
 					   line);
 				return -1;
 			}
+			return ret;
 		}
 
 		if (os_strncmp(pos, "wmm_ac_", 7) == 0) {
-- 
2.18.0

